src/main/scala/camp/task_1/Task1.scala

```diff
@@ -0,0 +1,18 @@
+package camp.task_1
+
+import scala.annotation.tailrec
+import scala.concurrent.duration.FiniteDuration
+
+class Task1 extends App {
+
+  @tailrec
+  final def retry[A](block: () => A,
+                     acceptResult: A => Boolean,
+                     retries: List[FiniteDuration]): A = {
+    val a = block.apply()
+    if (acceptResult.apply(a) || retries.isEmpty) return a
```

Comment:
return is used in scala in very very exceptional cases, I've never used it :) You can avoid it in this case for sure.